rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -185,6 +188,102 @@ __lockless_has_parameter(
   return parameters.find(name) != parameters.end();
 }
 
+// see https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon
+#define ULP 100
+RCLCPP_LOCAL
+bool
+__are_doubles_equal(double x, double y, int ulp = ULP)
```

Comment:
I know the example on cppreference uses an integer, but can `ulp` not be unsigned, e.g. `size_t`?

Reply:
Done in https://github.com/ros2/rclcpp/pull/735/commits/b8f48e1e6e629a3c6f8b912d9401bf1d7630fd60